libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -199,40 +210,41 @@ void DocStringTagParser::parseDocStrings(
 		else if (tagName == "return")
 		{
 			returnTagsVisited++;
+
+			vector<string> returnParameterNames;
+
 			if (auto const* varDecl = dynamic_cast<VariableDeclaration const*>(&_node))
 			{
 				solAssert(varDecl->isPublic(), "@return is only allowed on public state-variables.");
-				if (returnTagsVisited > 1)
-					m_errorReporter.docstringParsingError(
-						5256_error,
-						_node.documentation()->location(),
-						"Documentation tag \"@" + tagName + "\" is only allowed once on state-variables."
-					);
+				returnParameterNames = FunctionType(*varDecl).returnParameterNames();
```

Comment:
The `FunctionType()` c'tor requires existing type information which is only available after we ran the DeclarationChecker..